smac/utils/validate.py

```diff
@@ -0,0 +1,371 @@
+    """
+
+    def __init__(self, scenario, trajectory, output, rng=None):
+        """
+        Construct Validator for given scenario and trajectory.
+
+        Parameters
+        ----------
+        scenario: Scenario
+            scenario object for cutoff, instances and specifics
+        trajectory: Trajectory
+            trajectory to take incumbent(s) from
+        output: string
+            path to runhistory to be saved
+        rng: np.random.RandomState
+            Random number generator
+        """
+        self.logger = logging.getLogger(
+            self.__module__ + "." + self.__class__.__name__)
+
+        self.scen = scenario
+        self.traj = trajectory
+        if output:
+            self.output = output
+        else:
+            self.output =  "validation_rh.json"
+        if isinstance(rng, np.random.RandomState):
+            self.rng = rng
+        elif isinstance(rng, int):
+            self.rng = np.random.RandomState(seed=rng)
+        else:
+            num_run = np.random.randint(MAXINT)
+            self.rng = np.random.RandomState(seed=num_run)
+
+        self.rh = RunHistory(average_cost)  # update this rh with validation-runs
+
+    def validate(self, config_mode='def', instance_mode='test', repetitions=1,
+                 n_jobs=1, runhistory=None):
+        """
+        Validate configs on instances and save result in runhistory.
+
+        Parameters
+        ----------
+        config_mode: string
+            what configurations to validate
+            from [def, inc, def+inc, time, all], time means eval at 2^0, 2^1, 2^2, ...
+        instance_mode: string
+            what instances to use for validation, from [train, test, train+test]
+        repetitions: int
+            number of repetitions in nondeterministic algorithms
+        n_jobs: int
+            number of parallel processes used by joblib
+        runhistory: RunHistory or string or None
+            runhistory to take data from
+
+        Returns
+        -------
+        runhistory: RunHistory
+            runhistory with validated runs
+        """
+        # Reset runhistory
+        self.rh = RunHistory(average_cost)
+
+        # Get relevant configurations and instances
+        configs = self._get_configs(config_mode)
+        instances = self._get_instances(instance_mode)
+
+        # If runhistory is given as string, load into memory
+        if isinstance(runhistory, str):
+            fn = runhistory
+            runhistory = RunHistory(average_cost)
+            runhistory.load_json(fn, self.scen.cs)
+
+        # Get all runs needed as list
+        runs = self.get_runs(configs, instances, repetitions=repetitions,
+                             runhistory=runhistory)
+
+        # Create new Stats without limits
+        inf_scen = Scenario({'run_obj':'quality'})
+        inf_stats = Stats(inf_scen)
+        inf_stats.start_timing()
+
+        # Create TAE
+        tae = ExecuteTARunOld(ta=self.scen.ta,
```

Comment:
Would it be better to add an option to pass any instance of ExecuteTARun or do we specifically want to provide Old and AClib?